application/controllers/UserManagementController.php

```diff
@@ -762,7 +765,8 @@ public function actionImportUsers(string $importFormat = 'csv'): string
                 break;
             case "csv":
             default:
-                $aNewUsers = UserParser::getDataFromCSV($_FILES); //importFormat default is csv ...
+                $aNewUsers = UserParser::getDataFromCSV($_FILES);
+                $importFormat = 'csv'; /* Fix usage of importFormat in view */
```

Comment:
Probably this works,
Still, I believe the inteded behaviour is initializing the importFormat variable if it is not json nor CSV. right?
If that's the case, I feel the current implementation could be missleading/confusing.
I would feel a more clasic approach would be better.
`if (empty($importFormat)) $importFormat = 'csv'`

Then, also if no import format matches in the swithc, we could throw an error.
Why default is csv?

Reply:
>  if (empty($importFormat)) $importFormat = 'csv'

Not : must be 

`if( !in_array($importFormat,['csv','json']) {` because here importFormat came from URL (look at the issue)

At start i think i do such fix
```
if( !in_array($importFormat,['csv','json']) {
	throw new CHttpException(400);
}
```
But since previous version allow anything but force file to be csv (same after upload) : i allow it too.

You think it's better ?

Reply:
> I would say, to initilize the variable to 'csv' if it comes empty (as to have clean inside the function) and that's it. Just to avoid future XSS in this function if someone updates it.

I don't understand : if user create the url with userManagement/renderUserImport?importFormat=HACKED


You want to shown «Select HACKED file [file input]» ?

Can you explain what do you think it's better exacltly with example …



- userManagement/renderUserImport?importFormat=HACKED : Throw a 403 Yes or No
- userManagement/renderUserImport?importFormat=HACKED : show «Please choose HACKED file» Yes or No

Reply:
This is what I picture... makes sense to you?

Before line 760:
```
// Initialize $importFormat if not given. 
if (empty($importFormat)) {
   $importFormat = 'csv'
}
```

Then 
```
switch ($importFormat) {
    case "json":
        $aNewUsers = UserParser::getDataFromJSON($_FILES);
        break;
    case "csv":
        $aNewUsers = UserParser::getDataFromCSV($_FILES);
        break;
    default:
       // If no matching format, throw error
       throw new CHttpException(400);
}
```

About the error, maybe it is better to use LSUserxception?

Reply:
Initilizing is not needed : include in function,

Then i think a 

```
if( !in_array($importFormat,['csv','json']) {
	throw new LSUserxception(400);
}
```

is more clear.
I update only this.